File: Backend/UAM-LIFT/src/main/java/ni/edu/uam/UAM_LIFT/models/Carro.java
package ni.edu.uam.UAM_LIFT.models;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;


@Entity
@Table(name = "carros")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor

public class Carro {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;


    @Column(nullable = false, unique = true)
    private String placa;

    @Column(nullable = false)
    private String marca;

    @Column(nullable = false)
    private String modelo;

    @Column(nullable = false)
    private String color;

    // Relación: Muchos carros pueden pertenecer a un solo Usuario
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "usuario_id", nullable = false)
    @JsonProperty(access = JsonProperty.Access.WRITE_ONLY)
    private Usuario propietario;
}

File: Uam Lift/app/src/main/java/ni/edu/uam/uamlift/data/models/Carro.kt
package ni.edu.uam.uamlift.data.models

import com.google.gson.annotations.SerializedName

data class Carro(
    @SerializedName("id")
    val id: Long? = null,

    @SerializedName("placa")
    var placa: String,

    @SerializedName("marca")
    var marca: String,

    @SerializedName("modelo")
    var modelo: String,

    @SerializedName("color")
    var color: String,

    @SerializedName("propietario")
    var propietario: Usuario
)